### PR TITLE
e2e: Generate prometheus.yaml on setup

### LINKED ISSUE #899: e2e: Generate prometheus.yaml when Prometheus is enabled
### Summary

When Promethus is enabled in the manifest, create automatically a configuration file for Prometheus that includes essential settings and the IP addresses of the nodes in the manifest.

For example, for `test/e2e/networks/simple.toml`, which has 4 nodes, the content of the yaml file could be:
```
global:
  scrape_interval: 1s

scrape_configs:
  - job_name: 'validator01'
    static_configs:
      - targets: ['localhost:6701']

  - job_name: 'validator02'
    static_configs:
      - targets: ['localhost:6702']

  - job_name: 'validator03'
    static_configs:
      - targets: ['localhost:6703']

  - job_name: 'validator04'
    static_configs:
      - targets: ['localhost:6704']
```

### PR DESCRIPTION
**Pull request by hvanz**
Closes #899 

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by sergio-mena — APPROVED** [review]
LGTM

File: test/e2e/pkg/testnet.go [hunk 1]
```diff
@@ -482,6 +485,39 @@ func (t Testnet) HasPerturbations() bool {
...
+func (t Testnet) prometheusConfigBytes() ([]byte, error) {
+	tmpl, err := template.New("prometheus-yaml").Parse(`global:
+  scrape_interval: 1s
+
+scrape_configs:
+{{- range .Nodes }}
+  - job_name: '{{ .Name }}'
+    static_configs:
+      - targets: ['localhost:{{ .PrometheusProxyPort }}']
+{{end}}`)
```
Comment (mzabaluev) [inline test/e2e/pkg/testnet.go:497]: The template would be better put in a standalone file.

**Review by mzabaluev — APPROVED** [review]
Looks good with a minor nit pick.

File: test/e2e/pkg/testnet.go (hunk 1 above)
Comment (hvanz) [reply test/e2e/pkg/testnet.go]: Done. Thanks for the suggestion!

**Pull request merged**